src/main/java/at/hannibal2/skyhanni/features/inventory/ExperimentationTableDisplay.kt

```diff
@@ -0,0 +1,239 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.config.features.inventory.helper.HelperConfig.Experiments
+import at.hannibal2.skyhanni.events.GuiContainerEvent.BackgroundDrawnEvent
+import at.hannibal2.skyhanni.events.GuiContainerEvent.SlotClickEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object ExperimentationTableDisplay {
+
+    private val config get() = SkyHanniMod.feature.inventory.helper.enchanting
+
+    private var display = emptyList<String>()
+    private val startSlot = 9
+
+    private var uncoveredItems = mutableMapOf<Int, String>()
+    private var possiblePairs = 0
+
+    data class ItemPair(val first: Pair<Int, String>, val second: Pair<Int, String>)
+
+    private var foundNormals = mutableMapOf<Int, String>()
+    private var foundMatches = mutableListOf<ItemPair>()
+    private var foundPairs = mutableListOf<ItemPair>()
+    private var foundPowerUps = mutableMapOf<Int, String>()
+    private var toCheck = mutableListOf<Pair<Int, ItemStack>>()
+    private var lastClicked = mutableListOf<Int>()
+    private var currentExperiment = Experiments.NONE
+
+    private val patternGroup = RepoPattern.group("enchanting.experiments")
+    private val superpairsPattern by patternGroup.pattern(
+        "superpairs",
+        "Superpairs \\((?<experiment>.+)\\)")
+
+    private val powerUpPattern by patternGroup.pattern(
+        "powerups",
+        "Gained \\+\\d Clicks?|Instant Find|\\+\\S* XP")
+
+    private val rewardPattern by patternGroup.pattern(
+        "rewards",
+        "\\d{1,3}k Enchanting Exp|Enchanted Book|(?:Titanic |Grand |\\b)Experience Bottle|Metaphysical Serum|Experiment The Fish",
+    )
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        foundNormals.clear()
+        foundPairs.clear()
+        uncoveredItems.clear()
+        foundPowerUps.clear()
+        foundMatches.clear()
+        toCheck.clear()
+        lastClicked.clear()
+        display = emptyList()
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDrawn(event: BackgroundDrawnEvent) {
```

Comment:
Use ``GuiRenderEvent.ChestGuiOverlayRenderEvent`` instead of background drawnevent